bindings/nodejs/lib/types/block/address.ts

```diff
@@ -118,12 +133,101 @@ class NftAddress extends Address {
     }
 }
 
+/**
+ * An implicit account creation address.
+ */
+class ImplicitAccountCreationAddress extends Address {
+    /**
+     * The Ed25519 address.
+     */
+    readonly address: Ed25519Address;
```

Comment:
This is fine in Rust because we have a DTO but I'm not sure it's fine in JSON? The serialisation of that is supposed to be `pubKeyHash`, wouldn't this nest it in an address field?

Reply:
Oh, I just saw the toString sorry 🤦🏻

Reply:
Wait, will this actually be used for JSON ?

Reply:
youre missing @Type(() => Ed25519Address)

Reply:
That would handle the JSOn serialisation as if it was a Ed25519Address and not contained one?

Reply:
fixed it. FYI this is because the `class-transformer` lib requires nested classes to specify their types for...some reason (it's because javascript is a terrible language)

Reply:
I'd still like to have a confirmation from @kwek20 that this will skip the `address` key itself

Reply:
oh, no, that's not what it will do. I misunderstood the issue

Reply:
Either we find a serialisation trick, or we don't use a `Ed25519Address` but directly the inner field

Reply:
I vote for private inner field and a getter to get the Ed25519Address